### PR TITLE
Update Vitis AI EP to support multiple DPU targets through provider options

### PR DESCRIPTION
**Pull request by jtuyls**
Update the Vitis AI execution provider to support multiple DPU targets, specifically the Zynq DPUCZDX8G target (on arm64).

For documentation, do I need to create a separate PR on the gh-pages branch?

### CODE REVIEW

**Comment by faxu** [comment]
@jtuyls, yes please submit a PR for doc update on gh-pages branch.

**Comment by hariharans29** [comment]
Hi @jywu-msft : Can you please review this PR ?

**Comment by Elonaever** [comment]
so can I use the onnxruntime with DPUCVDX8G in the Versal AI Core board?

**Comment by jtuyls** [comment]
@Elonaever No, that DPU won't be supported yet. The specific DPU's that will be enabled are: DPUCZDX8G (Zynq) and DPUCAHX8H (U50/U280). However, the changes in this PR make it easier to add more DPU's later on.

**Review by skottmckay — APPROVED** [review]

**Comment by jtuyls** [comment]
@skottmckay I resolved the conflicts with master, could you have another look at this PR?

File: include/onnxruntime/core/providers/vitisai/vitisai_provider_factory.h [hunk 1]
```diff
@@ -10,7 +10,9 @@ extern "C" {
 /**
  * \param use_arena zero: false. non-zero: true.
  */
-ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_VITISAI, _In_ OrtSessionOptions* options, const char *backend_type, int device_id);
+ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_VITISAI, _In_ OrtSessionOptions* options,
+               const char* backend_type, int device_id, const char* export_runtime_module,
+               const char* load_runtime_module);
```
Comment (skottmckay) [inline include/onnxruntime/core/providers/vitisai/vitisai_provider_factory.h:16]: Might be worth putting the options in a struct as that's what we're doing for a number of other EPs now.

[https://github.com/microsoft/onnxruntime/blob/b854f2399dafb379584a0d7aebe42c958ae5aede/include/onnxruntime/core/session/onnxruntime_c_api.h#L269](https://github.com/microsoft/onnxruntime/blob/b854f2399dafb379584a0d7aebe42c958ae5aede/include/onnxruntime/core/session/onnxruntime_c_api.h#L269)

**Review by skottmckay — APPROVED** [review]
:shipit:

File: include/onnxruntime/core/providers/vitisai/vitisai_provider_factory.h (hunk 1 above)
Comment (jtuyls) [reply include/onnxruntime/core/providers/vitisai/vitisai_provider_factory.h:16]: Ok, you want me to do that in this PR still?

Comment (skottmckay) [reply include/onnxruntime/core/providers/vitisai/vitisai_provider_factory.h:16]: Whenever - not required in this PR. Just something that may grow better over time when you want to switch to that.

Comment (jtuyls) [reply include/onnxruntime/core/providers/vitisai/vitisai_provider_factory.h:16]: Ok, it looks like all tests have passed so maybe we merge the PR in like this and I will put all options in a struct in a future PR?

**Pull request merged**